### PR TITLE
Support SmoothQuant for ORT static quantization

### PR DESCRIPTION
**Pull request by mengniwang95**
### Description

Support SmoothQuant for ORT static quantization via intel neural compressor

> Note:
Please use neural-compressor==2.2 to try SmoothQuant function.

### Motivation and Context
For large language models (LLMs) with gigantic parameters, the systematic outliers make quantification of activations difficult. As a training free post-training quantization (PTQ) solution, SmoothQuant offline migrates this difficulty from activations to weights with a mathematically equivalent transformation. Integrating SmoothQuant into ORT quantization can benefit the accuracy of INT8 LLMs.

### CODE REVIEW

**Comment by hshen14** [comment]
Hi @yufenglee , please review the PR. Thanks.

**Comment by hshen14** [comment]
FYI - Intel Neural Compressor v2.2 was released with enhanced SmoothQuant. This PR enables SmoothQuant feature in ORT.

**Comment by mengniwang95** [comment]
@yufenglee hi, could you take a look and review this PR? thanks a lot 😄

**Review by yufenglee — APPROVED** [review]

**Comment by yufenglee** [comment]
@pranavsharma, could you please help review? The PR adds a package in requirements-dev.txt

**Review by pranavsharma — COMMENTED** [review]
This is a new dep. Have you followed the process of adding a new dep? See https://github.com/microsoft/onnxruntime/pull/15797#issuecomment-1535285188

**Comment by mengniwang95** [comment]
Hi, @pranavsharma  It seems the new dep neural-compressor isn't installed during CI test, could you provide me some help about where to add this dep for CI test?

File: requirements-dev.txt [hunk 1]
```diff
@@ -18,3 +18,4 @@ scipy
 sympy
 wheel
 setuptools>=41.4.0
+neural-compressor
```
Comment (yihonglyu) [inline requirements-dev.txt:21]: Doesn't neural-compressor need to greater or equal than a specific version like v2.2?

**Comment by hshen14** [comment]
> 

@yufenglee @yihonglyu  please it seems we don't have the permission. Please help trigger. Thx.

File: tools/ci_build/requirements.txt [hunk 2]
```diff
@@ -5,3 +5,6 @@ numpy==1.24.0
 coloredlogs==15.0
 transformers==4.24.0
 psutil
+
+# package used by smooth quant test
+neural-compressor>=2.2
```
Comment (tianleiwu) [inline tools/ci_build/requirements.txt:10]: Do not add it here. This is for transformer test.

For example, for Linux GPU pipeline, you can see 
https://github.com/microsoft/onnxruntime/blob/2ae041f390917f776b4bb746fe9570f047c274d9/tools/ci_build/github/azure-pipelines/linux-gpu-ci-pipeline.yml#L147

So need update here:
tools/ci_build/github/linux/docker/scripts/manylinux/requirements.txt

File: onnxruntime/test/python/quantization/test_quantize_static.py [hunk 3]
```diff
@@ -98,6 +98,19 @@ def test_static_quant_config(self):
         check_model_correctness(self, self._model_fp32_path, quant_model_path, data_reader.get_next())
         data_reader.rewind()
 
+    def test_smooth_quant(self):
```
Comment (tianleiwu) [inline onnxruntime/test/python/quantization/test_quantize_static.py:101]: I suggest to add a check that when neural-compressor is not installed, we can skip the test. In this way, we need not install neural-compressor and run this test in every pipeline.

Comment (hshen14) [reply onnxruntime/test/python/quantization/test_quantize_static.py:101]: Good point. Thanks for the suggestions.

Comment (mengniwang95) [reply onnxruntime/test/python/quantization/test_quantize_static.py:103]: I have updated my PR according to your suggestions, thanks a lot 😄

**Comment by mengniwang95** [comment]
Windows ARM64 build failed, but it seems to have nothing to do with INC. Could anyone help to take a look? Thanks a lot

**Comment by yufenglee** [comment]
@mengniwang95, does INC support ARM64? The ARM64 pipeline fails: https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1066017&view=logs&j=ceb7959a-7ebd-5187-d731-ecbb8f43bfff&t=6fcdf55b-f575-5bcb-f3eb-08852e21b1eb&l=42410

**Comment by tianleiwu** [comment]
> @mengniwang95, does INC support ARM64? The ARM64 pipeline fails: 

**ImportError: libGL.so.1**: cannot open shared object file: No such file or directory

May need install some dependent package like opencv-python-headless.

**Comment by hshen14** [comment]
> e dependent packag

Hi @tianleiwu, please help suggest the action how to proceed per this comment. Thanks.

**Comment by mengniwang95** [comment]
Could you help check why Windows_CI / Onnxruntime-TVM test failed? Thanks

**Comment by tianleiwu** [comment]
@mengniwang95, I re-run the failed pipelines.

**Comment by mengniwang95** [comment]
Hi, we install python3.8 (Use py3.8 because we find this '/home/cloudtest/.local/lib/python3.8/site-packages' in log file) in arm64v8/centos:7 container locally and reproduce the bug in arm CI test, after installing mesa-libGL that issue can be resolved. 
But even though I add mesa-libGL into https://github.com/microsoft/onnxruntime/blob/6e895fe70a227472a51e1f88f8d795234bb183d3/dockerfiles/Dockerfile.arm64#L17 the arm CI test still fails. So I wonder which OS does arm CI test use and where can I found the docker build info to ensure mesa-libGL is installed?

**Comment by tianleiwu** [comment]
@mengniwang95, the Linux CPU CI pipeline failed in running tools/ci_build/github/linux/run_python_tests.sh. 

I think it is not needed to update docker file. Try add dependent packages to requirements.txt that you've changed.

**Comment by mengniwang95** [comment]
@tianleiwu Hi, we released a new version which uses opencv-python-headless instead of opencv, we validated that opencv-python-headless can skip libGL.so.1 issue. Could you help tp trigger tests again?

**Comment by mengniwang95** [comment]
Hi, could someone help to trigger the test?

**Comment by mengniwang95** [comment]
Hi @yufenglee , I am not sure why ONNX Runtime React Native CI Pipeline failed, could you help to take a look?

**Comment by tianleiwu** [comment]
> Hi @yufenglee , I am not sure why ONNX Runtime React Native CI Pipeline failed, could you help to take a look?
The error seems not related to this change. I think it is fine. Merge main shall resolve the pipeline failure.

**Review by tianleiwu — APPROVED** [review]

**Comment by hshen14** [comment]
Thanks @yihonglyu @tianleiwu @yufenglee! Appreciate all the help during our first PR contribution to ORT.

**Pull request merged**